src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -1645,6 +1645,39 @@ public bool Equals(StringBuilder sb)
             }
         }
 
+        /// <summary>
+        /// Determines if the contents of this builder are equal to the contents of ReadOnlySpan<char>.
+        /// </summary>
+        /// <param name="value">The ReadOnlySpan<char>.</param>
+        public bool Equals(ReadOnlySpan<char> value)
+        {
+            if (value.Length != Length)
+                return false;
+
+            StringBuilder thisChunk = this;
+            int thisChunkIndex = thisChunk.m_ChunkLength;
+            var chunkChars = thisChunk.m_ChunkChars;
+
+            for (int i = Length - 1; i >= 0; i--)
```

Comment:
This look is very inefficient. The part that compares one character at a time needs to be a tight loop, perhaps just call `StringSpanHelpers.Equals(this ReadOnlySpan<char> left, ReadOnlySpan<char> right)`.

Reply:
But then it will convert the StringBuilder into string and then pass it to 
 ``` StringSpanHelpers.Equals(this ReadOnlySpan<char> left, ReadOnlySpan<char> right) ```
Wont this be inefficient if the strings are large ?

Reply:
Do you mean to call ```StringSpanHelpers.Equals(this ReadOnlySpan<char> left, ReadOnlySpan<char> right) ``` on the chunks of string (that stringbuilder allocates) ?

Reply:
Right.